app/post.go

```diff
@@ -448,6 +448,90 @@ func (a *App) UpdatePost(post *model.Post, safeUpdate bool) (*model.Post, *model
 	}
 }
 
+func (a *App) RethreadPost(post *model.Post, safeUpdate bool) (*model.Post, *model.AppError) {
+	post.SanitizeProps()
+
+	var oldPost *model.Post
+	var thread *model.PostList
+	if result := <-a.Srv.Store.Post().Get(post.Id); result.Err != nil {
+		return nil, result.Err
+	} else {
+		oldPost = result.Data.(*model.PostList).Posts[post.Id]
+		thread = result.Data.(*model.PostList)
+		if oldPost == nil {
+			err := model.NewAppError("UpdatePost", "api.post.rethread_post.find.app_error", nil, "id="+post.Id, http.StatusBadRequest)
+			return nil, err
+		}
+
+		if oldPost.DeleteAt != 0 {
+			err := model.NewAppError("UpdatePost", "api.post.rethread_post.permissions_details.app_error", map[string]interface{}{"PostId": post.Id}, "", http.StatusBadRequest)
+			return nil, err
+		}
+
+		if oldPost.IsSystemMessage() {
+			err := model.NewAppError("UpdatePost", "api.post.rethread_post.system_message.app_error", nil, "id="+post.Id, http.StatusBadRequest)
+			return nil, err
+		}
+	}
+
+	var originalRootId string
+	originalRootId = oldPost.RootId
+	newPost := &model.Post{}
+	*newPost = *oldPost
```

Comment:
Not your code (since we do this elsewhere), but this pattern is troublesome, since this isn't a deep copy -- both posts point at the same `Props`, `Filenames`, and `FileIds`. It all works out in this case.

Reply:
Should i do anything about this? I got my inspiration from updatePost when I made these.